index.js

```diff
@@ -2,23 +2,50 @@
 const pathExists = require('path-exists');
 const modifyFilename = require('modify-filename');
 
-const incrementer = filePath => {
-	let counter = 0;
-	return () => modifyFilename(filePath, (filename, extension) => `${filename} (${++counter})${extension}`);
+const parenthesesIncrementer = (filename, extension) => {
+	let [, originalFilename, index] = filename.match(/^(.*)\((\d+)\)$/) || [null, filename, 0];
+	return `${originalFilename.trim()} (${++index})${extension}`;
 };
 
-const unusedFilename = filePath => {
-	const getFilePath = incrementer(filePath);
-	const find = async newFilePath => await pathExists(newFilePath) ? find(getFilePath()) : newFilePath;
-	return find(filePath);
+const separatorIncrementer = separator => (filename, extension) => {
+	let [, originalFilename, index] = new RegExp(`^(.*)${separator}(\\d+)$`).exec(filename) || [null, filename, 0];
```

Comment:
Use named capture groups for the regex.

Reply:
I did want to do that, but that would break backwards compatibility with current `package.json:engines` (`>=8`).

Not an issue?

Reply:
Seeing as end of life for node 10 was 6 days ago, I think I'll just bump it. Complain if I should revert it.